packages/backend-function/src/schedule_parser.ts

```diff
@@ -0,0 +1,209 @@
+import { CronOptions, Rule, Schedule } from 'aws-cdk-lib/aws-events';
+import { NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs';
+import * as targets from 'aws-cdk-lib/aws-events-targets';
+import { AmplifyUserError } from '@aws-amplify/platform-core';
+import type { Cron, Rate, TimeInterval } from './factory.js';
+
+/**
+ * Parses function schedule props in order to create EventBridge rules.
+ */
+export class ScheduleParser {
+  /**
+   * Initialize EventBridge rules
+   */
+  constructor(
```

Comment:
non-blocker: since this class is purely internal it's not a big deal, but this is a lot of logic for a ctor. I think this is a case where we may not even need a class and just a `convertTimeIntervalsToEventRules` function would suffice for this functionality.

I also think the caller should be responsible for attaching the event rule to the lambda rather than it happening internally here. It's a small single-responsibility violation.